logic/tool/TypeResolver.java

```diff
@@ -86,10 +101,10 @@ public Conjunction resolveLabels(Conjunction conjunction) {
         return conjunction;
     }
 
-    public Conjunction resolve(Conjunction conjunction) {
+    public Conjunction resolve(Conjunction conjunction, boolean insertable) {
         resolveLabels(conjunction);
-        TraversalBuilder traversalConstructor = new TraversalBuilder(conjunction, conceptMgr);
-        Map<Reference, Set<Label>> resolvedLabels = executeResolverTraversals(traversalConstructor);
+        TraversalBuilder traversalBuilder = new TraversalBuilder(conjunction, conceptMgr, insertable);
+        Map<Reference, Set<Label>> resolvedLabels = executeResolverTraversals(traversalBuilder);
         if (resolvedLabels.isEmpty()) {
             conjunction.setSatisfiable(false);
```

Comment:
since we set conjunctions as satisfiable now, instead of variables do we: 
1. at all need the isSatisfiable flag in Variable?
2. in rule validation, do we not have to check the conjunction's `isSatisfiable` flag instead of each variable's?

Reply:
Discussed with Josh, we will park this issue as it needs more architectural thought.